applications/ChimeraApplication/custom_strategies/strategies/fs_strategy_for_chimera.h

```diff
@@ -0,0 +1,762 @@
+            (*iExtraSteps)->Execute();
+
+        const double stop_solve_time = OpenMPUtils::GetCurrentTime();
+        KRATOS_INFO_IF("FSStrategyForChimera", (BaseType::GetEchoLevel() >= 1 && r_model_part.GetCommunicator().MyPID() == 0)) << "Time for solving step : " << stop_solve_time - start_solve_time << std::endl;
+
+        return norm_dp;
+    }
+
+
+
+    void ComputeSplitOssProjections(ModelPart& rModelPart) override
+    {
+        const array_1d<double,3> zero(3,0.0);
+
+        array_1d<double,3> out(3,0.0);
+
+#pragma omp parallel
+        {
+            ModelPart::NodeIterator nodes_begin;
+            ModelPart::NodeIterator nodes_end;
+            OpenMPUtils::PartitionedIterators(rModelPart.Nodes(),nodes_begin,nodes_end);
+
+            for ( ModelPart::NodeIterator it_node = nodes_begin; it_node != nodes_end; ++it_node )
+            {
+                it_node->FastGetSolutionStepValue(CONV_PROJ) = zero;
+                it_node->FastGetSolutionStepValue(PRESS_PROJ) = zero;
+                it_node->FastGetSolutionStepValue(DIVPROJ) = 0.0;
+                it_node->FastGetSolutionStepValue(NODAL_AREA) = 0.0;
+            }
+        }
+
+#pragma omp parallel
+        {
+            ModelPart::ElementIterator elem_begin;
+            ModelPart::ElementIterator elem_end;
+            OpenMPUtils::PartitionedIterators(rModelPart.Elements(),elem_begin,elem_end);
+
+            for ( ModelPart::ElementIterator it_elem = elem_begin; it_elem != elem_end; ++it_elem )
+            {
+                it_elem->Calculate(CONV_PROJ,out,rModelPart.GetProcessInfo());
+            }
+        }
+
+        rModelPart.GetCommunicator().AssembleCurrentData(CONV_PROJ);
+        rModelPart.GetCommunicator().AssembleCurrentData(PRESS_PROJ);
+        rModelPart.GetCommunicator().AssembleCurrentData(DIVPROJ);
+        rModelPart.GetCommunicator().AssembleCurrentData(NODAL_AREA);
+
+        // If there are periodic conditions, add contributions from both sides to the periodic nodes
+        //PeriodicConditionProjectionCorrection(rModelPart);
+        ChimeraProjectionCorrection(rModelPart);
+#pragma omp parallel
+        {
+            ModelPart::NodeIterator nodes_begin;
+            ModelPart::NodeIterator nodes_end;
+            OpenMPUtils::PartitionedIterators(rModelPart.Nodes(),nodes_begin,nodes_end);
+
+            for ( ModelPart::NodeIterator it_node = nodes_begin; it_node != nodes_end; ++it_node )
+            {
+                const double nodal_area = it_node->FastGetSolutionStepValue(NODAL_AREA);
+                if( nodal_area > mAreaTolerance )
+                {
+                    it_node->FastGetSolutionStepValue(CONV_PROJ) /= nodal_area;
+                    it_node->FastGetSolutionStepValue(PRESS_PROJ) /= nodal_area;
+                    it_node->FastGetSolutionStepValue(DIVPROJ) /= nodal_area;
+                } else {
+                    KRATOS_WARNING_ONCE("Nodal area too small ! ")<<std::endl;
```

Comment:
this kills the parallelism!
=> should not be used inside omp-loops

Reply:
This was a debug left over ! Removed !